docs/testing-tool.rst

```diff
@@ -7,85 +7,125 @@ Use the KSQL testing tool to test a set of KSQL statements. The KSQL testing too
-          ],
-          "inputs": [
-            {"topic": "input_topic", "key": 0, "value": {"id": 8}, "timestamp": 0},
-            {"topic": "input_topic", "key": 0, "value": {"id": 10}, "timestamp": 10000},
-            {"topic": "input_topic", "key": 1, "value": {"id": 9}, "timestamp": 30000},
-            {"topic": "input_topic", "key": 1, "value": {"id": 11}, "timestamp": 40000}
-          ],
-          "outputs": [
-            {"topic": "OUTPUT", "key": 0, "value": {"ID": 8}, "timestamp": 0},
-            {"topic": "OUTPUT", "key": 1, "value": {"ID": 9}, "timestamp": 30000}
-          ]
-        },
-        {
-          "name": "my first negative test",
-          "description": "an example negative test where the statement will fail to parse",
-          "statements": [
-            "CREATE STREAM TEST WITH (kafka_topic='test_topic', value_format='DELIMITED');"
-          ],
-          "expectedException": {
-            "type": "io.confluent.ksql.util.KsqlException",
-            "message": "The statement does not define any columns."
-          }
-        }
-      ]
-    }
+Statements File
+---------------
 
+The statements file contain the KSQL statements that will be tested. The following are the supported statements in the testing tool:
 
-You can have multiple tests in one test file. In addition to name, description, and statements, each test includes
-input topics and their data along with the expected output topics and their data.
-The test file format is the same as the files KSQL code uses for integration tests. For more details on the
-structure of the test file and all possible settings, see the `README.md <https://github.com/confluentinc/ksql/tree/master/ksql-functional-tests>` in the KSQL repository.
+- CREATE STREAM
+- CREATE TABLE
+- CREATE STREAM AS SELECT
+- CREATE TABLE AS SELECT
+- INSERT INTO SELECT
 
-Running Tests
-*************
+Here is a sample statements file for testing tool:
 
-Assume we run the previous test file, which is stored in test.json in the home directory.
-The following command shows how to run the test from the terminal:
+.. code:: sql
+    CREATE STREAM TEST (ID bigint, NAME varchar, VALUE bigint) WITH (kafka_topic='left_topic', value_format='JSON', key='ID');
+    CREATE STREAM TEST_STREAM (ID bigint, F1 varchar, F2 bigint) WITH (kafka_topic='right_topic', value_format='JSON', key='ID');
+    CREATE STREAM LEFT_OUTER_JOIN as SELECT t.id, name, value, f1, f2 FROM test t left join TEST_STREAM tt WITHIN 11 seconds ON t.id = tt.id;
+    CREATE STREAM foo AS SELECT t_id, name FROM LEFT_OUTER_JOIN WHERE t_id = 90;
+    CREATE STREAM bar AS SELECT * FROM foo;
 
-.. code:: bash
+Input File
+----------
+
+The input file is a JSON file with one array field named "inputs". Each element in the array is the representation of input messages.
+A message should have topic, key, value and timestamp. The following is a sample input file for the above test:
+
+.. code:: json
+    {
+      "inputs": [
```

Comment:
Do we need to wrap the inputs and outputs in JSON objects like this, now that each gets its own dedicated file? It feels odd that we currently require inputs and outputs to be wrapped, while the statements file shouldn't be.

If we chose to unwrap the inputs and outputs, that would also address @agavra 's nit about `InputRecordsNode.java` and `OutputRecordsNode.java` being identical minus a name.

Reply:
This indeed came up in my discussion with MD too. The reason we wrap them in a JSON object is to have the possibility to add more fields in future.